### PR TITLE
T-49 Fix code smells

### PR DESCRIPTION
**Pull request by DashaDrahun**
Run resharper code analyzer on solution. Fixed discovered issues. Contradictive types of issues were written in file.

### CODE REVIEW

File: PortfolioKeeper/BusinessLogic/Services/AssetService.cs [hunk 1]
```diff
@@ -130,13 +131,10 @@ public AssetDto GetAsset(int assetId)
 
             var asset = assetRepository.GetById(assetId);
 
-            if (asset == null)
-            {
-                logger.Warn($"Asset is null. AssetService.GetAssetById assetId: {assetId}");
-                throw new ArgumentException($"Asset not found, id: {assetId}");
-            }
+            if (asset != null) return mapper.Map<Asset, AssetDto>(asset);
+            logger.Warn($"Asset is null. AssetService.GetAssetById assetId: {assetId}");
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Services/AssetService.cs]: add empty line between if and logger

File: PortfolioKeeper/BusinessLogic/Services/AssetService.cs [hunk 2]
```diff
@@ -196,12 +194,7 @@ public void Create(List<AssetPropertyValueViewModel> properties, string userId,
             }
 
             var portfolio = portfolioRepository.Get(x => x.AppUser.Id == userId).FirstOrDefault();
-
-            if (portfolio == null)
-            {
-                throw new NullReferenceException("Portfolio can't be null");
-            }
-
+            if (portfolio == null) throw new NullReferenceException("Portfolio can't be null");
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Services/AssetService.cs]: move throw to new line. add empty line after

File: PortfolioKeeper/BusinessLogic/Services/SecurityCheckService.cs [hunk 3]
```diff
@@ -33,11 +33,9 @@ public void CheckPortfolioAccess(int portfolioId, string userId)
                 throw new ArgumentException($"Portfolio not found, id: {portfolioId}");
             }
 
-            if (portfolio.AppUser.Id != userId)
-            {
-                logger.Warn($"No permissions. SecurityCheckService.CheckPortfolioAccess assetId: {portfolioId} userId:{userId}");
-                throw new SecurityException($"No permission to access the portfolio portfolioId:{portfolioId}");
-            }
+            if (portfolio.AppUser.Id == userId) return;
+            logger.Warn($"No permissions. SecurityCheckService.CheckPortfolioAccess assetId: {portfolioId} userId:{userId}");
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Services/SecurityCheckService.cs]: empty line between if and logger

File: PortfolioKeeper/BusinessLogic/Services/SecurityCheckService.cs [hunk 4]
```diff
@@ -50,11 +48,9 @@ public void CheckAssetAccess(int assetId, string userId)
                 throw new ArgumentException($"Portfolio not found, id: {assetId}");
             }
 
-            if (asset.Portfolio.AppUser.Id != userId)
-            {
-                logger.Warn($"No permissions. SecurityCheckService.CheckAssetAccess assetId: {assetId} userId:{userId}");
-                throw new SecurityException($"No permission to access the asset portfolioId:{assetId}");
-            }
+            if (asset.Portfolio.AppUser.Id == userId) return;
+            logger.Warn($"No permissions. SecurityCheckService.CheckAssetAccess assetId: {assetId} userId:{userId}");
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Services/SecurityCheckService.cs]: empty line

File: PortfolioKeeper/Contracts/Enums/CurrencyValue.cs [hunk 5]
```diff
@@ -1,4 +1,5 @@
-﻿namespace PortfolioKeeper.Contracts.Enums
+﻿// ReSharper disable InconsistentNaming
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Contracts/Enums/CurrencyValue.cs]: ? What resharper proposes here?

File: PortfolioKeeper/Controllers/AccountController.cs [hunk 6]
```diff
@@ -63,7 +64,9 @@ public ActionResult Login(string returnUrl)
             }
 
             var result = await signInManager.PasswordSignInAsync(model.UserName,
-                model.Password, model.RememberMe, shouldLockout: true);
+                model.Password,
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Controllers/AccountController.cs]: leave like it was - two lines

File: PortfolioKeeper/Controllers/AccountController.cs [hunk 7]
```diff
@@ -125,20 +128,22 @@ public ActionResult Register()
             var callbackUrl = Url.Action("ConfirmEmail", "Account", new { userId = user.Id, code }, protocol: Request?.Url?.Scheme);
 
             await userManager.SendEmailAsync(user.Id,
-                "Confirm your account", "Please confirm your account by clicking <a href=\"" + callbackUrl + "\">here</a>");
+                "Confirm your account",
+                "Please confirm your account by clicking <a href=\"" + callbackUrl + "\">here</a>");
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Controllers/AccountController.cs]: revert

File: PortfolioKeeper/Controllers/AccountController.cs [hunk 8]
```diff
@@ -125,20 +128,22 @@ public ActionResult Register()
...
+                Email = model.Email
             };
 
             return View("EmailNotConfirmed", null, newModel);
         }
 
         [AllowAnonymous]
-        public async Task<ActionResult> ConfirmEmail(string userId, string code)
+        public async Task<ActionResult> ConfirmEmail(string userId,
+            string code)
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Controllers/AccountController.cs]: revert

File: PortfolioKeeper/Controllers/AccountController.cs [hunk 9]
```diff
@@ -219,14 +225,11 @@ public ActionResult ResetPassword(string code)
 
             var result = await userManager.ResetPasswordAsync(user.Id, model.Code, model.Password);
 
-            if (!result.Succeeded)
-            {
-                AddErrors(result);
+            if (result.Succeeded) return RedirectToAction("ResetPasswordConfirmation", "Account");
+            AddErrors(result);
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Controllers/AccountController.cs]: empty line

File: PortfolioKeeper/Controllers/ManageController.cs [hunk 10]
```diff
@@ -153,14 +153,12 @@ public ActionResult SetPassword()
             user.LastName = model.LastName;
             var result = await userManager.UpdateAsync(user);
 
-            if (!result.Succeeded)
-            {
-                AddErrors(result);
+            if (result.Succeeded)
+                return RedirectToAction("Index", new {Message = ManageMessageId.ChangeProfileSuccess});
+            AddErrors(result);
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Controllers/ManageController.cs]: empty line

File: PortfolioKeeper/Views/Manage/Index.cshtml [hunk 11]
```diff
@@ -13,14 +13,8 @@
...
-            @if (Model.HasPassword)
-            {
-                @Html.ActionLink("Change your password", "ChangePassword")
-            }
-            else
-            {
-                @Html.ActionLink("Create", "SetPassword")
-            }
+            @(Model.HasPassword ? Html.ActionLink("Change your password", "ChangePassword") 
+                  : Html.ActionLink("Create", "SetPassword"))
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Views/Manage/Index.cshtml]: @(Model.HasPassword ? 
                                    Html.ActionLink("Change your password", "ChangePassword") 
                                    : Html.ActionLink("Create", "SetPassword"))

File: PortfolioKeeper/BusinessLogic/Services/SecurityCheckService.cs [hunk 12]
```diff
@@ -50,11 +48,9 @@ public void CheckAssetAccess(int assetId, string userId)
...
+            throw new SecurityException($"No permission to access the asset portfolioId:{assetId}");
```
Comment (goremukin) [inline PortfolioKeeper/BusinessLogic/Services/SecurityCheckService.cs]: Likewise

File: PortfolioKeeper/Controllers/AccountController.cs (hunk 6 above)
Comment (goremukin) [inline PortfolioKeeper/Controllers/AccountController.cs]: move `model.RememberMe,` and `shouldLockout: true);` to `model.Password,` line:
```
var result = await signInManager.PasswordSignInAsync(model.UserName,
                model.Password, model.RememberMe, shouldLockout: true);
```

File: PortfolioKeeper/Controllers/AccountController.cs (hunk 7 above)
Comment (goremukin) [inline PortfolioKeeper/Controllers/AccountController.cs]: move this to the previous line

File: PortfolioKeeper/Controllers/AccountController.cs [hunk 13]
```diff
@@ -183,7 +188,8 @@ public ActionResult ForgotPassword()
             var callbackUrl = Url.Action("ResetPassword", "Account", new { userId = user.Id, code }, Request?.Url?.Scheme);
 
             await userManager.SendEmailAsync(user.Id,
-                "Reset Password", "Please reset your password by clicking <a href=\"" + callbackUrl + "\">here</a>");
+                "Reset Password",
+                "Please reset your password by clicking <a href=\"" + callbackUrl + "\">here</a>");
```
Comment (goremukin) [inline PortfolioKeeper/Controllers/AccountController.cs]: move this to the previous line

File: PortfolioKeeper/Controllers/AccountController.cs [hunk 14]
```diff
@@ -219,14 +225,11 @@ public ActionResult ResetPassword(string code)
 
             var result = await userManager.ResetPasswordAsync(user.Id, model.Code, model.Password);
 
-            if (!result.Succeeded)
-            {
-                AddErrors(result);
+            if (result.Succeeded) return RedirectToAction("ResetPasswordConfirmation", "Account");
```
Comment (goremukin) [inline PortfolioKeeper/Controllers/AccountController.cs]: Likewise

File: PortfolioKeeper/Controllers/ManageController.cs [hunk 15]
```diff
@@ -153,14 +153,12 @@ public ActionResult SetPassword()
             user.LastName = model.LastName;
             var result = await userManager.UpdateAsync(user);
 
-            if (!result.Succeeded)
-            {
-                AddErrors(result);
+            if (result.Succeeded)
```
Comment (goremukin) [inline PortfolioKeeper/Controllers/ManageController.cs]: Likewise

File: PortfolioKeeper/Data/Repositories/UnitOfWork.cs [hunk 16]
```diff
@@ -37,14 +37,12 @@ public void Dispose()
 
         protected virtual void Dispose(bool disposing)
         {
-            if (!disposed)
+            if (disposed) return;
```
Comment (goremukin) [inline PortfolioKeeper/Data/Repositories/UnitOfWork.cs]: Likewise

File: PortfolioKeeper/Data/Repositories/UnitOfWork.cs [hunk 17]
```diff
@@ -37,14 +37,12 @@ public void Dispose()
...
+            if (disposing)
```
Comment (goremukin) [inline PortfolioKeeper/Data/Repositories/UnitOfWork.cs]: Likewise

File: PortfolioKeeper/Helpers/AddingAssetHelper.cs [hunk 18]
```diff
@@ -64,31 +63,17 @@ public CarViewModel GetCar(List<AssetPropertyTypeDto> allPropertyTypes)
...
         {
-            var property = src.GetType().GetProperty(propName);
-
-            if (property != null)
-            {
-                return property.GetValue(src, null);
-            }
-
-            return null;
+            return src.GetType().GetProperty(propName)?.GetValue(src, null);
```
Comment (goremukin) [inline PortfolioKeeper/Helpers/AddingAssetHelper.cs]:
```
var property = src.GetType().GetProperty(propName);

            if (property != null)
            {
                return property.GetValue(src, null);
            }

            return null;
```

File: PortfolioKeeper/Helpers/AddingAssetHelper.cs [hunk 19]
```diff
@@ -64,31 +63,17 @@ public CarViewModel GetCar(List<AssetPropertyTypeDto> allPropertyTypes)
...
         {
-            var first = allPropertyTypes.FirstOrDefault(e => propertyName != null && e.Name == propertyName);
-
-            if (first != null)
-            {
-                return first.DisplayName;
-            }
-
-            return null;
+            return allPropertyTypes.FirstOrDefault(e => propertyName != null && e.Name == propertyName)?.DisplayName;
```
Comment (goremukin) [inline PortfolioKeeper/Helpers/AddingAssetHelper.cs]:
```
var first = allPropertyTypes.FirstOrDefault(e => propertyName != null && e.Name == propertyName);

            if (first != null)
            {
                return first.DisplayName;
            }
```

File: PortfolioKeeper/Views/Account/Login.cshtml [hunk 20]
```diff
@@ -7,7 +7,7 @@
 <div class="row">
     <div class="col-md-8">
         <section id="loginForm">
-            @using (Html.BeginForm("Login", "Account", new { ReturnUrl = ViewBag.ReturnUrl }, FormMethod.Post, new { @class = "form-horizontal", role = "form" }))
+            @using (Html.BeginForm("Login", "Account", new {ViewBag.ReturnUrl }, FormMethod.Post, new { @class = "form-horizontal", role = "form" }))
```
Comment (goremukin) [inline PortfolioKeeper/Views/Account/Login.cshtml]: Add space:
`new { ViewBag.ReturnUrl }`

File: PortfolioKeeper/Contracts/Enums/CurrencyValue.cs (hunk 5 above)
Comment (DashaDrahun) [reply PortfolioKeeper/Contracts/Enums/CurrencyValue.cs]: The question is Currency enum with uppercase members (BYN, USD). Resharper sugggest Byn, and i disabled it by comment

**Comment by DashaDrahun** [comment]
Corrected your review issues

File: PortfolioKeeper/BusinessLogic/Services/AssetService.cs [hunk 21]
```diff
@@ -194,7 +196,9 @@ public void Create(List<AssetPropertyValueViewModel> properties, string userId,
             }
 
             var portfolio = portfolioRepository.Get(x => x.AppUser.Id == userId).FirstOrDefault();
-            if (portfolio == null) throw new NullReferenceException("Portfolio can't be null");
+            if (portfolio == null)
+                throw new NullReferenceException("Portfolio can't be null");
+
             Asset asset = new Asset()
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Services/AssetService.cs]: () - remove please

**Pull request merged**